### PR TITLE
PYTHON-3786 add types to srv_resolver.py

### PR DESCRIPTION
**Pull request by sleepyStick**
https://jira.mongodb.org/browse/PYTHON-3786

### CODE REVIEW

File: pymongo/srv_resolver.py [hunk 1]
```diff
@@ -27,17 +29,18 @@
 from pymongo.common import CONNECT_TIMEOUT
 from pymongo.errors import ConfigurationError
 
-
 # dnspython can return bytes or str from various parts
 # of its API depending on version. We always want str.
-def maybe_decode(text):
+
+
```
Comment (ShaneHarvey) [inline pymongo/srv_resolver.py:35]: Could you undo the whitespace changes here?

File: pymongo/srv_resolver.py [hunk 2]
```diff
@@ -117,10 +128,10 @@ def _get_srv_response_and_hosts(self, encapsulate_errors):
...
-    def get_hosts(self):
+    def get_hosts(self) -> List[Tuple[str, Any]]:
         _, nodes = self._get_srv_response_and_hosts(True)
         return nodes
 
-    def get_hosts_and_min_ttl(self):
+    def get_hosts_and_min_ttl(self) -> Tuple[List[Tuple[str, Any]], int]:
         results, nodes = self._get_srv_response_and_hosts(False)
-        return nodes, results.rrset.ttl
+        return nodes, results.rrset.ttl  # type: ignore[union-attr]
```
[the remark this answers is not in the document]
Comment (sleepyStick) [reply pymongo/srv_resolver.py:136]: results.rrset is an `Optional[RRset]`, and in the case that its none, it won't have a `ttl`. Is there a better way to handle this?

[the remark this answers is not in the document]
Comment (ShaneHarvey) [reply pymongo/srv_resolver.py:136]: Only thing I can think of is:
```python
rrset = results.rrset
ttl = rrset.ttl if rrset else 0
return nodes, ttl 
```

[the remark this answers is not in the document]
Comment (sleepyStick) [reply pymongo/srv_resolver.py]: Works for me. Changed.

**Pull request merged**